pkg/command/service/load.go

```diff
@@ -0,0 +1,423 @@
+	serviceLoadCommand.Flags().DurationVarP(&loadArgs.WaitPodsReadyDuration, "wait-time", "w", 10*time.Second, "Time to wait for all pods to be ready")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.LoadTool, "load-tool", "t", "wrk", "Select the load test tool, support wrk and hey")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.LoadConcurrency, "load-concurrency", "c", "30", "total number of workers to run concurrently for the load test tool")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.LoadDuration, "load-duration", "d", "60s", "Duration of the test for the load test tool")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.Output, "output", "o", ".", "Measure result location")
+
+	return serviceLoadCommand
+}
+
+func LoadServicesUpFromZero(params *pkg.PerfParams, inputs pkg.LoadArgs) error {
+	ctx := context.Background()
+	nsNameList, err := GetNamespaces(ctx, params, inputs.Namespace, inputs.NamespaceRange, inputs.NamespacePrefix)
+	if err != nil {
+		return err
+	}
+	loadFromZeroResult, err := loadAndMeasure(ctx, params, inputs, nsNameList, getServices)
+	if err != nil {
+		return err
+	}
+
+	knativeVersion := GetKnativeVersion(params)
+	ingressInfo := GetIngressController(params)
+	loadFromZeroResult.KnativeInfo.ServingVersion = knativeVersion["serving"]
+	loadFromZeroResult.KnativeInfo.EventingVersion = knativeVersion["eventing"]
+	loadFromZeroResult.KnativeInfo.IngressController = ingressInfo["ingressController"]
+	loadFromZeroResult.KnativeInfo.IngressVersion = ingressInfo["version"]
+
+	rows := make([][]string, 0) // replicas ready duration of all services
+
+	maxReplicasCount, replicasCountList := getReplicasCount(loadFromZeroResult)
+
+	//Add replicas ready duration results of each service to rows, only select results before the replicas reaches the maximum count
+	for i := 0; i < len(loadFromZeroResult.Measurment); i++ {
+		var row []string
+		row = append(row, loadFromZeroResult.Measurment[i].ServiceName, loadFromZeroResult.Measurment[i].ServiceNamespace)
+		for j := 0; j < len(loadFromZeroResult.Measurment[i].ReplicaResults); j++ {
+			if loadFromZeroResult.Measurment[i].ReplicaResults[j].ReadyReplicasCount <= replicasCountList[i] {
+				row = append(row, strconv.FormatFloat(loadFromZeroResult.Measurment[i].ReplicaResults[j].ReplicaReadyDuration, 'f', 3, 32))
+			}
+		}
+		rows = append(rows, row)
+	}
+
+	sortSlice(rows)
+
+	//Add the column name
+	var row []string
+	row = append(row, "svc_name", "svc_namespace")
+	for i := 1; i <= maxReplicasCount; i++ {
+		row = append(row, "replica_"+strconv.Itoa(i)+"_ready")
+	}
+	rows = append([][]string{row}, rows...)
+
+	current := time.Now()
+	outputLocation, err := utils.CheckOutputLocation(inputs.Output)
+	if err != nil {
+		fmt.Printf("failed to check measure output location: %s\n", err)
+	}
+
+	csvPath := filepath.Join(outputLocation, fmt.Sprintf("%s_%s.csv", current.Format(DateFormatString), LoadOutputFilename))
```

Comment:
These look fairly similar to what's done in `measure.go`, so would it make sense to consolidate some of these output methods into `common.go`? (possibly in a follow-on PR)

Reply:
Good idea! I can try it in the follow-on PR.